low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -529,84 +579,103 @@ def _generate_cpu_data(self):
-                                self._node_sensor.iowait_time,
-                                self._node_sensor.nice_time,
-                                self._node_sensor.softirq_time,
-                                self._node_sensor.steal_time,
-                                self._node_sensor.system_time,
-                                self._node_sensor.user_time,
-                                self._node_sensor.cpu_core_data,
-                                self._node_sensor.cpu_usage,
-                                self.FAULT,
-                                fault_event
-                            )
-
-            # Add in uuid if it was present in the json request
-            if self._uuid is not None:
-                cpuDataMsg.set_uuid(self._uuid)
-            jsonMsg = cpuDataMsg.getJson()
-            self.cpu_sensor_data = jsonMsg
-            self.os_sensor_type["cpu_usage"] = self.cpu_sensor_data
-
-            # Transmit it to message processor
-            self._write_internal_msgQ(EgressProcessor.name(), jsonMsg)
-            # Store the state to Persistent Cache.
-            self.persist_state_data('cpu', 'CPU_USAGE_DATA')
-
-        if self._node_sensor.cpu_usage < self._cpu_usage_threshold \
-           and self.high_usage['cpu']:
-
-            # Create the cpu usage data message and hand it over to the egress processor to transmit
-            fault_resolved_event = "CPU usage decreased to %s, lesser than configured threshold of %s" \
-                %(self._node_sensor.cpu_usage, self._cpu_usage_threshold)
-            logger.warning(fault_resolved_event)
-
-            # Create the local mount data message and hand it over to the egress processor to transmit
-            cpuDataMsg = CPUdataMsg(self._node_sensor.host_id,
-                                self._epoch_time,
-                                self._node_sensor.csps,
-                                self._node_sensor.idle_time,
-                                self._node_sensor.interrupt_time,
-                                self._node_sensor.iowait_time,
-                                self._node_sensor.nice_time,
-                                self._node_sensor.softirq_time,
-                                self._node_sensor.steal_time,
-                                self._node_sensor.system_time,
-                                self._node_sensor.user_time,
-                                self._node_sensor.cpu_core_data,
-                                self._node_sensor.cpu_usage,
-                                self.FAULT_RESOLVED,
-                                fault_resolved_event
-                            )
+        cpu_persistent_data = self.read_persistent_data('CPU_USAGE_DATA')
+        if cpu_persistent_data['cpu_usage_time_map']:
+            previous_check_time = int(cpu_persistent_data['cpu_usage_time_map'])
+        else:
+            previous_check_time = int(-1)

```

Comment:
same comment as memory, why not setting current_time here itself?

Reply:
Please check the response for memory